mvlearn/decomposition/grouppca.py

```diff
@@ -0,0 +1,358 @@
+        for X, mean in zip(Xs, self.individual_mean_):
+            lstq_solution = np.dot(transformed_pinv, X - mean)
+            self.individual_projections_.append(linalg.pinv(lstq_solution).T)
+            self.individual_embeddings_.append(lstq_solution.T)
+        self.components_ = pca.components_
+        self.explained_variance_ = pca.explained_variance_
+        self.explained_variance_ratio_ = pca.explained_variance_ratio_
+        return self
+
+    def transform(self, Xs, y=None):
+        r"""Apply groupPCA to Xs.
+
+        Xs is projected on the principal components learned
+        in the training set.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : list of array-likes or numpy.ndarray
+            The transformed data.
+            If `multiple_outputs` is True, it is a list with the estimated
+            individual sources.
+            If `multiple_outputs` is False, it is a single array containing the
+            shared sources.
+        """
+        check_is_fitted(self)
+        Xs = check_Xs(Xs, copy=True)
+        if self.multiple_outputs:
+            return [
+                np.dot(X - mean, W.T)
+                for W, X, mean in (
+                    zip(
+                        self.individual_projections_, Xs, self.individual_mean_
+                    )
+                )
+            ]
+        if self.individual_pca_:
+            for i, (X, mean, components_, explained_variance_) in enumerate(
+                zip(
+                    Xs,
+                    self.individual_mean_,
+                    self.individual_components_,
+                    self.individual_explained_variance_,
+                )
+            ):
+                X = X - mean
+                X_transformed = np.dot(X, components_.T)
+                if self.prewhiten:
+                    X_transformed /= np.sqrt(explained_variance_)
+                Xs[i] = X_transformed
+        else:
+            Xs = [X - mean for X, mean in zip(Xs, self.individual_mean_)]
+        X_stack = np.hstack(Xs)
+        X_transformed = np.dot(X_stack, self.components_.T)
+        if self.whiten:
+            X_transformed /= np.sqrt(self.explained_variance_)
+        return X_transformed
+
+    def inverse_transform(self, X_transformed):
+        r"""
+        Recover multiview data from transformed data.
+
+        Returns an array Xs such that the transform of Xs would be
+        X_transformed
+
+        Parameters
+        ----------
+        X_transformed : list of array-likes or numpy.ndarray
+            The dataset corresponding to transformed data
+
+        Returns
+        -------
+        Xs : list of arrays
+            The recovered individual datasets
+        """
+        check_is_fitted(self)
```

Comment:
I think it makes sense to `check_Xs` the `X_transformed` here.

Reply:
I added it if multiple_outputs=True. Otherwise, the input X_transformed should be a plain 2d array, so no need to call check_Xs on it right?

Reply:
Yes you are right